kafka-native/src/main/java/org/ballerinalang/messaging/kafka/utils/KafkaUtils.java

```diff
@@ -840,4 +819,22 @@ public void notifyFailure(BError error) {
         }
         return result;
     }
+
+    public static Type getAttachedFunctionReturnType(BObject serviceObject, String functionName, int paramCount) {
+        MethodType function = null;
+        MethodType[] resourceFunctions = serviceObject.getType().getMethods();
+        for (MethodType resourceFunction : resourceFunctions) {
+            if (functionName.equals(resourceFunction.getName())) {
+                function = resourceFunction;
+                Type[] parameterTypes = function.getParameterTypes();
+                if (parameterTypes.length == paramCount) {
+                    Type returnType = function.getReturnType();
+                    return returnType;
+                } else {
+                    throw createKafkaError("invalid " + functionName + " remote function signature");
```

Comment:
I don't think this is a good idea. The method signatures are validated by the compiler plugin. Not sure if this will get thrown for other non-remote functions the service might have?.